test/utils_test.js

```diff
@@ -431,6 +432,60 @@ describe("utils", () => {
       expect(mergeObjects(obj1, obj2)).eql(expected);
     });
 
+    describe("trimObject()", () => {
+      it("should't mutate the provided objects", () => {
+        const obj = { a: undefined };
+        trimObject(obj);
+        expect(obj).eql({ a: undefined });
+      });
+
+      it("should trim empty object", () => {
+        const obj = {};
+        const res = trimObject(obj);
+        expect(res).eql(undefined);
+      });
+
+      it("should trim object with no fields set", () => {
+        const obj = { a: undefined, b: undefined, c: undefined };
+        const res = trimObject(obj);
+        expect(res).eql(undefined);
+      });
+
+      it("shouldn't trim object with fields set", () => {
+        const obj = { a: true, b: undefined, c: undefined };
+        const res = trimObject(obj);
+        expect(res).eql({ a: true, b: undefined, c: undefined });
+      });
+
+      it("should trim field object", () => {
+        const obj = { a: true, o: { b: undefined } };
+        const res = trimObject(obj);
+        expect(res).eql({ a: true, o: undefined });
+      });
+
+      it("shouldn't trim deeply nested object with field set", () => {
+        const obj = { a: true, o: { b: { c: true } } };
+        const res = trimObject(obj);
+        expect(res).eql({ a: true, o: { b: { c: true } } });
+      });
+
+      it("should trim multiple object", () => {
+        const obj = {
+          a: true,
+          o1: { b: { c: undefined }, d: undefined },
+          o2: { e: undefined },
+        };
+        const res = trimObject(obj);
+        expect(res).eql({ a: true, o1: undefined, o2: undefined });
+      });
+
+      it("shouldn't trim object with empty array field", () => {
+        const obj = { a: [] };
+        const res = trimObject(obj);
+        expect(res).eql({ a: [] });
+      });
+    });
```

Comment:
I'd like to see more functional tests added at the component level for this behavior.